### DISCUSSION TITLE
Cannot Get Spiral Simulation Working

### DISCUSSION THREAD

**Discussion by nathanjbeaumont** [MRsources/MRzero-Core#78] [Q&A] [answered]
Hello all,
I am trying to get a simple spiral sequence simulation working but the signal always comes out as all 0's. The sequence is a uniform density spiral with a 90deg tip at the start. I have run this same sequence in the KomaMRI simulator and JERMIS and the signal output looks fine in those two. Does anyone have an idea of what's going wrong? Attached is the a zip so you can recreate.
[mrZeroSpiral.zip](https://github.com/user-attachments/files/17608068/mrZeroSpiral.zip)
![image](https://github.com/user-attachments/assets/cbaeb4f2-9296-4daf-81c9-6a9f7b78a8a3)

**Comment by mzaiss** [comment]
So spirals are working see an example file here for pypulseq 1.3.1

https://colab.research.google.com/github/MRsources/MRzero-Core/blob/main/documentation/playground_mr0/mr0_spiral_GRE.ipynb

it is hard to say from your seq file only,
but it seems your slice selection gradient and the rewinder gradient have
the same area, but the rewinder should have half the area. thus the signal
is spoiled.

try removing all slice selection gradients first, if this gives signal the
issue is slice selection

may i ask how many isochromats you used in jemris or koma?

**Reply by mzaiss** [reply, marked as answer]
Yes, its your slice selection.
![Unbenannt](https://github.com/user-attachments/assets/0b5ccb10-a948-4f50-8217-38a7d16c893b)
![Unbenannt](https://github.com/user-attachments/assets/9af888a7-bd8e-4183-8c8a-e068862f4a68)

gzr too large or gz too small.

gradient moment in z not zero during the spiral:

![Unbenannt-1](https://github.com/user-attachments/assets/ad533570-aaed-42b8-b672-e79d96243abf)

this sequence wont give signal at a real scanner.

why it gives signal in Jemris or Koma is that you probably used too few isochromats, so they will just get a phase, but the spoiling is not properly simulated, you might need 1000 isochromats or so, then the signal would also cancel to zero there.

**Reply by nathanjbeaumont** [reply]
Oh of course, you're correct, it works fine if the slice select rewinder area is halved.

**Reply by mzaiss** [reply]
how many spins did you use in koma or jemris?

**Reply by nathanjbeaumont** [reply]
I'm unsure. I used the brain_phantom2D phantom in KomaMRI, and the provided 2D brain phantom in the JEMRIS_sim Matlab GUI.
